frost/bip340_test.go

```diff
@@ -407,6 +409,219 @@ func TestBip340CiphersuiteHash(t *testing.T) {
+			publicKeyX: "D69C3509BB99E412E68B0FE8544E72837DFA30746D8BE2AA65975F29D22DC7B9",
+			message:    "4DF3C3F68FCC83B27E9D42C90431A72499F17875C81A599B566C9889B9696703",
+			isValid:    true,
+		},
+		{
+			signature:   "6CFF5C3BA86C69EA4B7376F31A9BCB4F74C1976089B2D9963DA2E5543E17776969E89B4C5564D00349106B8497785DD7D1D713A8AE82B32FA79D5F7FC407D39B",
+			publicKeyX:  "EEFDEA4CDB677750A420FEE807EACF21EB9898AE79B9768766E4FAA04A2D4A34",
+			message:     "243F6A8885A308D313198A2E03707344A4093822299F31D0082EFA98EC4E6C89",
+			isValid:     false,
+			expectedErr: "point publicKey is infinite",
+		},
+
+		{
+			signature:   "FFF97BD5755EEEA420453A14355235D382F6472F8568A18B2F057A14602975563CC27944640AC607CD107AE10923D9EF7A73C643E166BE5EBEAFA34B1AC553E2",
+			publicKeyX:  "DFF1D77F2A671C5F36183726DB2341BE58FEAE1DA2DECED843240F7B502BA659",
+			message:     "243F6A8885A308D313198A2E03707344A4093822299F31D0082EFA98EC4E6C89",
+			isValid:     false,
+			expectedErr: "coordinate R.y is not even",
+		},
+		{
+			signature:   "1FA62E331EDBC21C394792D2AB1100A7B432B013DF3F6FF4F99FCB33E0E1515F28890B3EDB6E7189B630448B515CE4F8622A954CFE545735AAEA5134FCCDB2BD",
+			publicKeyX:  "DFF1D77F2A671C5F36183726DB2341BE58FEAE1DA2DECED843240F7B502BA659",
+			message:     "243F6A8885A308D313198A2E03707344A4093822299F31D0082EFA98EC4E6C89",
+			isValid:     false,
+			expectedErr: "coordinate R.y is not even",
+		},
+
+		{
+			signature:   "6CFF5C3BA86C69EA4B7376F31A9BCB4F74C1976089B2D9963DA2E5543E177769961764B3AA9B2FFCB6EF947B6887A226E8D7C93E00C5ED0C1834FF0D0C2E6DA6",
+			publicKeyX:  "DFF1D77F2A671C5F36183726DB2341BE58FEAE1DA2DECED843240F7B502BA659",
+			message:     "243F6A8885A308D313198A2E03707344A4093822299F31D0082EFA98EC4E6C89",
+			isValid:     false,
+			expectedErr: "coordinate R.x != r",
+		},
+		{
+			signature:   "0000000000000000000000000000000000000000000000000000000000000000123DDA8328AF9C23A94C1FEECFD123BA4FB73476F0D594DCB65C6425BD186051",
+			publicKeyX:  "DFF1D77F2A671C5F36183726DB2341BE58FEAE1DA2DECED843240F7B502BA659",
+			message:     "243F6A8885A308D313198A2E03707344A4093822299F31D0082EFA98EC4E6C89",
+			isValid:     false,
+			expectedErr: "point R is infinite",
+		},
+		{
+			signature:   "00000000000000000000000000000000000000000000000000000000000000017615FBAF5AE28864013C099742DEADB4DBA87F11AC6754F93780D5A1837CF197",
+			publicKeyX:  "DFF1D77F2A671C5F36183726DB2341BE58FEAE1DA2DECED843240F7B502BA659",
+			message:     "243F6A8885A308D313198A2E03707344A4093822299F31D0082EFA98EC4E6C89",
+			isValid:     false,
+			expectedErr: "point R is infinite",
+		},
+		{
+			signature:   "4A298DACAE57395A15D0795DDBFD1DCB564DA82B0F269BC70A74F8220429BA1D69E89B4C5564D00349106B8497785DD7D1D713A8AE82B32FA79D5F7FC407D39B",
+			publicKeyX:  "DFF1D77F2A671C5F36183726DB2341BE58FEAE1DA2DECED843240F7B502BA659",
+			message:     "243F6A8885A308D313198A2E03707344A4093822299F31D0082EFA98EC4E6C89",
+			isValid:     false,
+			expectedErr: "point R is infinite",
+		},
```

Comment:
This test vector is problematic. In BIP-340 test vectors, [the comment is](https://github.com/bitcoin/bips/blob/deae64bfd31f6938253c05392aa355bf6d7e7605/bip-0340/test-vectors.csv#L13):

> sig[0:32] is not an X coordinate on the curve

So I would expect the test vector to fail on "point R is infinite" check. It does fail later on "coordinate R.x != r". I checked and the point seems to be a valid point on the curve with `[x,y]` as
[101293062680523315514373137351023114440902235251657644508821325047911886333529, 95491709537915294920828256998521669146617750390665870859237534620269297521559]

Reply:
The signature verification algorithm in [BIP-340](https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki#verification) does not check that the 32 public key bytes in the signature (`rb` in the code, `r` in the BIP) match a valid point on the curve. This check is effectively implicit in the algorithm later, because `G` and `P` are known valid curve points, so the point `R` that is calculated is also a valid point. Thus, if `r` is not a valid point's X-coordinate, `R.X != r`.

Reply:
Ah, that makes sense! Fixed in 929f34ddccefd1a357804b1dfa2004db4ee4b4c4.